user/api.go

```diff
@@ -376,12 +417,14 @@ func (a *Api) UpdateUser(res http.ResponseWriter, req *http.Request, vars map[st
 				}
 			}
 
-			if updatedUser.EmailVerified && updatedUser.TermsAccepted != "" {
+			if updatedUser.EmailVerified && updatedUser.TermsAccepted != "" && !strings.HasSuffix(updatedUser.Username, "@tidepool.io") && !strings.HasSuffix(updatedUser.Username, "@tidepool.org") {
 				if a.marketoManager != nil && a.marketoManager.IsAvailable() {
-					if updateUserDetails.EmailVerified != nil || updateUserDetails.TermsAccepted != nil {
-						a.marketoManager.CreateListMembershipForUser(updatedUser)
+					if  updateUserDetails.EmailVerified != nil || updateUserDetails.TermsAccepted != nil {
+						a.logger.Printf("Sending create Kafka message for: %v", updatedUser.Id)
+						a.KafkaProducer("create-user", updatedUser.Id)
```

Comment:
Instead of implementing message sending here, I think it's better if you create a new service `KafkaMarketoManager` which implements the `marketo.Manager` interface. This new service will then produce a kafka message.

Reply:
I have it here (or at least the call of KafkaProducer) because I want a marketo message triggered every-time a user is fully created (when terms are accepted) or account is updated. Is there another way to do this? If you are talking about the functions themselves I will put them into their own folder like I did for platform.

let me know if this is not what you mean

Reply:
I'd publish the updated user and update details object itself (with redacted password) and do this check in the marketo service.